lib/elixir_bench/benchmarks/benchmarks.ex

```diff
@@ -51,6 +51,16 @@ defmodule ElixirBench.Benchmarks do
     Repo.fetch(where(Job, uuid: ^uuid))
   end
 
+  def fetch_running_job_by_reference(branch_name, commit_sha) do
+    query =
+      Job
+      |> last
+      |> Job.unfinished()
+      |> where(branch_name: ^branch_name, commit_sha: ^commit_sha)
```

Comment:
where does the `reference` come from? I wonder if it's an appropriate name for it.

I also think we can just drop the branch name - commit shas shouldn't duplicate so we're save there. This seems to be missing any scoping by repository or so which we should rather introduce imo

Reply:
reference I meant `branch name` and `commit`... but you are right, this is not the best name. I will also add a check for the repo as I said in my previous comment.  

About skipping the branch name, I agree that it's possible and simpler. But I like the idea of separate builds for each branch to maintain the history and not just respond with the previous results.. But then we think about evolving this feature.

Reply:
added the scope and changed to a more sigificative name